### ISSUE TITLE
Persist operation does not unwrap StorageException

### ISSUE THREAD

**Issue by Scooletz** [Particular/NServiceBus.Persistence.AzureStorage#196]
Currently, Persist operation does not unwraps properly instances of `Microsoft.WindowsAzure.Storage.StorageException`. 

https://github.com/Particular/NServiceBus.Persistence.AzureStorage/blob/a73d380e6de3367ea032d38dd4bfba61b8cb1734/src/NServiceBus.Persistence.AzureStorage/SagaPersisters/AzureSagaPersister.cs#L174

In case of an optimistic concurrency exception, the message that is logged is a bit opaque

```
Microsoft.WindowsAzure.Storage.StorageException: Element 0 in the batch returned an unexpected response code.
```

**Comment by yvesgoeleven** [comment]
@Scooletz @HEskandari there is no other information to unwrap, storage service doesn't tell us what is wrong.

**Comment by Scooletz** [comment]
@yvesgoeleven Maybe my phrasing was unfortunate. What I meant was that, currently, we're bubbling up the storage exception with the cryptic message. Maybe we should `catch` it and provide a more meaningful exception showing what was wrong (for instance the optimistic concurrency etc). According to the [Azure doco](https://docs.microsoft.com/en-us/rest/api/storageservices/performing-entity-group-transactions#batch-response-syntax), a batch operation provides a detailed response, and as I vaguely remember, this can be accessed via some property of the `StorageException`. 

Summing up, the issue was about throwing a better exception, that can be logged in a more user-friendly way and provide some not-so-cryptic information

**Comment by yvesgoeleven** [comment]
> According to the Azure doco, a batch operation provides a detailed response, and as I vaguely remember, this can be accessed via some property of the StorageException.

Docs are lying, at least in the past there was no such info in the exception object. If you put fiddler in between you can get some additional information from the underlying http body response, but even that is often the same message for different error cases. You cannot (or at least could not) distinguish e.g. a wrong character in the partition key from an illegal date (anything before januari 1 1601) or an missing etag.

I have no issue with investigating if this improved over time, just don't want to get your hopes up ;)

**Comment by SeanFeldman** [comment]
> I have no issue with investigating if this improved over time, just don't want to get your hopes up ;)

Would raising this with the product group save us lengthy discussions about what can and cannot be done? Just saying 🙂

**Comment by Scooletz** [comment]
If this the statement about disability to retrieve error is right, this issue might be closed. My initial statement was about the disability to figure out what's wrong, when seeing cryptic message in logs:

> Microsoft.WindowsAzure.Storage.StorageException: Element 0 in the batch returned an unexpected response code.

**Comment by SeanFeldman** [comment]
This package is based on the old Storage SDK which has not been changed. The exception still the same exception.
